src/Appwrite/Platform/Modules/Functions/Workers/Builds.php

```diff
@@ -713,6 +720,107 @@ protected function buildDeployment(Device $deviceForFunctions, Func $queueForFun
 
+            if ($resource->getCollection() === 'sites') {
+                try {
+                    $rule = Authorization::skip(fn () => $dbForPlatform->findOne('rules', [
+                        Query::equal("projectInternalId", [$project->getInternalId()]),
+                        Query::equal("resourceType", ["deployment"]),
+                        Query::equal("resourceInternalId", [$deployment->getInternalId()])
+                    ]));
+
+                    if ($rule->isEmpty()) {
+                        throw new \Exception("Rule for build not found");
+                    }
+
+                    $client = new FetchClient();
+                    $client->addHeader('content-type', FetchClient::CONTENT_TYPE_APPLICATION_JSON);
+
+                    $bucket = $dbForPlatform->getDocument('buckets', 'screenshots');
+
+                    $configs = [
+                        'screenshot' => [
+                            'headers' => [ 'x-appwrite-hostname' => $rule->getAttribute('domain') ],
+                            'url' => 'http://traefik/?appwrite-preview=1&appwrite-theme=light',
+                            'theme' => 'light'
+                        ],
+                        'screenshotDark' => [
+                            'headers' => [ 'x-appwrite-hostname' => $rule->getAttribute('domain') ],
+                            'url' => 'http://traefik/?appwrite-preview=1&appwrite-theme=dark',
+                            'theme' => 'dark'
+                        ],
+                    ];
+
+                    $jwtObj = new JWT(System::getEnv('_APP_OPENSSL_KEY_V1'), 'HS256', 900, 0);
+                    $apiKey = $jwtObj->encode([
+                        'hostnameOverride' => true
+                    ]);
+
+                    // TODO: @Meldiron if becomes too slow, do concurrently
+                    foreach ($configs as $key => $config) {
+                        $config['headers'] = \array_merge($config['headers'] ?? [], [
+                            'x-appwrite-key' => API_KEY_DYNAMIC . '_' . $apiKey
+                        ]);
+
+                        $response = $client->fetch(
+                            url: 'http://appwrite-browser:3000/v1/screenshots',
+                            method: 'POST',
+                            body: $config
+                        );
+
+                        if ($response->getStatusCode() >= 400) {
+                            throw new \Exception("Screenshot failed to generate: " . $response->getBody());
+                        }
+
+                        $screenshot = $response->getBody();
+
+                        $fileId = ID::unique();
+                        $fileName = $fileId . '.png';
+                        $path = $deviceForFiles->getPath($fileName);
+                        $path = str_ireplace($deviceForFiles->getRoot(), $deviceForFiles->getRoot() . DIRECTORY_SEPARATOR . $bucket->getId(), $path); // Add bucket id to path after root
```

Comment:
Why do we have to do this? it feels like this should be simpler, or abstracted somehow, no?

Reply:
We do this in storage endpoint for createFile. I just copied it. We can improve it, agree, but let's do it in different scope from Sites.

Reply:
See if we can have something in storage backlog, and we can give it some love later. I would be very specific about the task as it will be hard to recall later what we wanted to achieve here.

Reply:
Noted down